### PR TITLE
Replace unit tests with Localstack for AWS integrations

### LINKED ISSUE #2127: Consider Testcontainers (Localstack) to replace Mockito for AWS integrations
**Is your feature request related to a problem? Please describe.**
Not a problem but rather making tests run against real integration.

**Describe the solution you'd like**
Currently, all of AWS Repository tests are being mocked with Mockito, [Localstack](https://www.testcontainers.org/modules/localstack/) can be used to provide real integration tests. This would help when new features or integrations are introduced.

**Additional context**
If this is something that is worth doing, I am willing to take this issue.

### PR DESCRIPTION
**Pull request by MatejNedic**
Closes #2127

### CODE REVIEW

File: spring-cloud-config-server/pom.xml [hunk 1]
```diff
@@ -202,6 +202,18 @@
...
 			<scope>test</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.testcontainers</groupId>
+			<artifactId>localstack</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<!-- AWS SDK v1 is required by testcontainers-localstack -->
+		<dependency>
+			<groupId>com.amazonaws</groupId>
```
Comment (ryanjbaxter) [inline spring-cloud-config-server/pom.xml:212]: Is this OK even though we are using v2?

Comment (eddumelendez) [reply spring-cloud-config-server/pom.xml:212]: yes, it is ok. Testcontainers Localstack module has a dependency on aws sdk v1 within the container class `LocalStackContainer`. It should be removed in Testcontainers 2.0

**Comment by ryanjbaxter** [comment]
@MatejNedic can you resolve the merge conflicts?

**Comment by ryanjbaxter** [comment]
@MatejNedic sorry one more thing, can you merge the latest changes from main into your branch?

**Comment by MatejNedic** [comment]
Hey @ryanjbaxter, I merged the latest changes from the main to my branch.

**Comment by MatejNedic** [comment]
I see build is failing due checkstyle. Will resolve that today.

**Comment by MatejNedic** [comment]
Hey @ryanjbaxter , sorry for a delay. I have pushed checkstyle fix.

**Comment by ryanjbaxter** [comment]
Looks like there are some test failures, see the GitHub build

**Comment by MatejNedic** [comment]
Pushed the fix. Locally tests are passing now for all changed classes.

**Pull request merged**